### PR TITLE
Neat 334 asset loader graph store interface

### PR DESCRIPTION
**Pull request by nikokaoja**
### Added
- `AssetLoader` added to load assets to CDF
- `get_default_prefixes` method to provide default prefixes
### Removed
- `PREFIXES` dict that was used as default prefixes in `NeatGraphStore` and `Rules`
### Improved
- `AssetRules` properties have now mandatory `implementation` field
### Fixed
- Issue of properties not being renamed in `DESCRIBE` query

### CODE REVIEW

File: tests/tests_unit/graph/test_loaders/test_asset_loader.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+from cognite.neat.graph.stores import NeatGraphStore
+from cognite.neat.rules.importers import ExcelImporter
+from tests import config
+
+
+def test_generation_of_assets():
+    store = NeatGraphStore.from_memory_store()
+    store.write(RdfFileExtractor(nordic44_knowledge_graph, base_uri=URIRef("http://purl.org/nordic44#")))
+
+    asset_rules, _ = ExcelImporter(filepath=config.DATA_FOLDER / "asset-architect-test.xlsx").to_rules()
```
Comment (doctrino) [inline tests/tests_unit/graph/test_loaders/test_asset_loader.py:16]: Based on the most recent failure of the coverage tests failing: https://github.com/cognitedata/neat/actions/runs/9971226811/job/27551846145?pr=542

It looks like the `asset_architect_test.xlsx` is not providing valid results thus making asset_rules None. Suggest to make this
```suggestion
    asset_rules = ExcelImporter(filepath=config.DATA_FOLDER / "asset-architect-test.xlsx").to_rules(errors="raise")
```

Then you should get better error messages at least.

File: cognite/neat/constants.py [hunk 2]
```diff
@@ -14,15 +14,17 @@
...
+def get_default_prefixes() -> dict[str, Namespace]:
+    return {
+        "rdf": RDF._NS,
+        "rdfs": RDFS._NS,
+        "dct": DCTERMS._NS,
+        "skos": SKOS._NS,
+        "owl": OWL._NS,
+        "xsd": XSD._NS,
+        "pav": Namespace("http://purl.org/pav/"),
+    }
```
Comment (doctrino) [inline cognite/neat/constants.py:27]: Nice!

File: cognite/neat/graph/loaders/_rdf2asset.py [hunk 3]
```diff
@@ -104,6 +134,39 @@ def categorize_assets(self, client: CogniteClient) -> None:
 
         raise NotImplementedError("Not implemented yet, this is placeholder")
 
+    def _load(self, stop_on_exception: bool = False) -> Iterable[AssetWrite | NeatIssue]:
```
Comment (doctrino) [inline cognite/neat/graph/loaders/_rdf2asset.py:137]: This is such a nice method. Really like how easy it is with the `store.read`

**Review by doctrino — APPROVED** [review]
Just a few minor comments. 

One suggestion: What about renaming it to `_rdf2class` and then have the loader create both Assets and Relationships.

File: cognite/neat/graph/loaders/_rdf2asset.py [hunk 4]
```diff
@@ -84,7 +107,14 @@ def from_rules(
...
         return cls(
-            rules, graph_store, data_set_id, use_orphanage, use_labels, asset_external_id_prefix, metadata_keys, issues
+            graph_store,
+            rules,
+            data_set_id,
+            use_orphanage,
+            use_labels,
+            asset_external_id_prefix,
+            metadata_keys,
+            issues,
```
[the remark this answers is not in the document]
Comment (nikokaoja) [reply cognite/neat/graph/loaders/_rdf2asset.py:117]: :) agree, it is pretty much useless, unless we do as you suggested (pass data_set_external_id

**Pull request merged**